charts/linkerd-control-plane/values.yaml

```diff
@@ -158,8 +158,11 @@ proxy:
   # for more info on container lifecycle hooks.
   waitBeforeExitSeconds: 0
   #  -- If set, the application container will not start until the proxy is
-  # ready
+  # ready. Only applicable to injected proxies
   await: true
+  #  -- If set, the control plane container will not start until the proxy is
+  # ready. Only applicable to destination and proxy-injector proxies
+  controlPlaneAwait: true
```

Comment:
Do we really need a separate config for this? What problems would arise if we used the existing config to control the control plane's configuration?

Reply:
I put it as a separate config as using the existing one would alter the behavior of existing deployments, and users would need to be aware of the change to the behavior of this value. Also, some users may desire the injected applications to not have the lifecycle hook but want the hooks on the control plane or are ok with the longer spin-up time that the lifecycle hook can cause. I figured it would be better to be non-breaking while being a little more verbose.

Reply:
@lamebear hey, just out of curiosity, what do you mean by:

> using the existing one would alter the behavior of existing deployments

The `await` flag generally applies to any deployment, except the control plane's. If we would change this flag, on upgrades, the control plane manifests would be rendered without the `postStart` hook, but only if the value is overridden. e.g `linkerd upgrade --set "await=false"`, in which case it'd be an informed decision for the operator? 

I'm not sure the PR would break anything: people who upgrade can now make the choice of changing the control plane manifests to not include the `postStart` hook, and pods will no longer contain the hook when re-deployed. Setting the value to `false` implies that existing pods won't have the hook, so I don't see this as having any side effects. The only "surprise" here would be for people that have been running with `await=false` who will (on upgrade) no longer have the hook on the control plane components -- this shouldn't break anything though, right?

> Also, some users may desire the injected applications to not have the lifecycle hook but want the hooks on the control plane or are ok with the longer spin-up time that the lifecycle hook can cause

Hmm, that's a fair point. This can already be done through annotations though. https://github.com/linkerd/linkerd2/pull/5967#issue-843556278 has more details, but I believe people can already opt-out (or opt-in, if it's off by default) by specifying `config.linkerd.io/await` on any namespace and/or pod. You wouldn't be able to add the annotation on the control plane, since it doesn't pass through the injector but it makes sense to me to have the control plane respect whatever default value you've chosen to go with.

In my mind, you wouldn't want the hooks when:
* Lifecycle hooks don't play well with the rest of your environment (e.g CNI plugins that use `NetworkPolicy` resources seem to have issues with lifecycle hooks);
* You want all of your containers to be started in parallel.

It feels more intuitive to me to just have one value that configures all meshed pods to behave in a certain way, with the ability to override this on a pod-by-pod basis if you need to. It also introduces less configuration. Just my thoughts though :) If having just one flag is a legitimate problem, I'm fine with introducing an additional one.